### ISSUE TITLE
Regression (2.2.2 and 2.3) for conjunctive condition on object value

### ISSUE THREAD

**Issue by WolfgangFahl** [SemanticMediaWiki/SemanticMediaWiki#1362]
The following query (see https://www.semantic-mediawiki.org/wiki/User:WolfgangFahl/Workdocumentation_2016-01-11_WF)

will behave differently in SMW 2.2.2 and 2.3:

```{{#ask: [[Wolfgang Fahl]] [[Attendee name::!Fahl]] [[Attendee publish::true]]
|? Attendee name
|? Attendee publish
}}```

In 2.2.2 the query will not return a result in 2.3 it does return a result (which IMHO is wrong).
My understanding is that query conditions are by default calling for an "AND" clause between the different conditions.

see
https://www.semantic-mediawiki.org/wiki/Help:Selecting_pages

So the above query calls for Wolfgang Fahl's page to be selected but it may not have the name "Fahl" and it should have the pusblish field true. Of these three conditions one is false so there should be no result.

**Comment by mwjames** [comment]
> [[Wolfgang Fahl]] [[Attendee name::!Fahl]] [[Attendee publish::true]]

I think if you try `[[Attendee name::!Fahl]] [[Attendee publish::true]] [[Wolfgang Fahl]]` then it should work as expected __BUT__ this is caused by #19 and was implemented with #1060.

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/4bcfea210409ff6113348b42af7271a353618024

**Comment by WolfgangFahl** [comment]
@mwjames  Thanx - I have added your query proposal to https://www.semantic-mediawiki.org/wiki/User:WolfgangFahl/Workdocumentation_2016-01-11_WF and it behaves as you asumed. For me this is a vaild workaround. As outlined in #19 some testcases would IMHO be the way to go to avoid such regressions for the future.  Hopefully I'll get a better grip on creating PhpUnit Testcase for SMW soon so that I can actively supply input.

**Comment by mwjames** [comment]
#1363 should restore the query plan for `[[Wolfgang Fahl]] [[Attendee name::!Fahl]] [[Attendee publish::true]]` independent of its conjunctional position.

> . As outlined in #19 some testcases would IMHO be the way to go to avoid such regressions for the future.

We had some test cases [0] but apparently this fall though the cracks so having a large set of queries is crucial in fighting possible regressions.

> creating PhpUnit Testcase for SMW soon so that I can actively supply input.

In this particular case it is not so much about PhpUnit Testcase or coverage ([1] is covered with 99%) the issue here boils down to the integration with an actual DB and QueryEngine (with either SQL or SPARQL) therefore a better venue in terms of cost/benefit is to have integration tests [2].

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Integration/ByJsonScript/Fixtures/q-0901.json

[1] https://scrutinizer-ci.com/g/SemanticMediaWiki/SemanticMediaWiki/code-structure/master/class/SMW%5CSQLStore%5CQueryEngine%5CQuerySegmentListProcessor

[2] https://www.semantic-mediawiki.org/wiki/Help:Integration_tests

**Comment by mwjames** [comment]
Fixed with c60a6d2
